packages/falcon-server-env/src/cache/Cache.ts

```diff
@@ -40,51 +42,20 @@ export default class Cache<V = any> implements KeyValueCache<V> {
 
   /**
    * Returns cached value for the provided key and setOptions object
-   * @param {string} key Cache key
-   * @param {GetCacheOptions} setOptions Object with params to fetch the data to be cached
-   * @returns {Promise<string|undefined>} Cached value
+   * @param key Cache key
+   * @param setOptions Object with params to fetch the data to be cached
+   * @returns Cached value
    */
   async get(key: string, setOptions?: GetCacheOptions): Promise<V> {
-    let value: GetCacheFetchResult = await this.cacheProvider.get(key);
-    try {
-      value = JSON.parse(value);
-    } catch {
-      // Keep `value` with the original value
-    }
-
-    // Validating by cache tags
-    if (this.isValueWithOptions(value)) {
-      const { tags: tagMap = {} } = value.options as ValueOptions;
-      if (await this.isTagMapValid(tagMap as TagMap)) {
-        ({ value } = value);
-      } else {
-        // If tags are invalid - set value as "not found"
-        value = undefined;
-        await this.delete(key);
-      }
+    if (this.getMap.has(key)) {
+      return this.getMap.get(key) as Promise<V>;
     }
 
-    if (typeof value === 'undefined' && typeof setOptions === 'object') {
-      const { fetchData } = setOptions;
-      let { options } = setOptions;
-      const cacheResult: GetCacheFetchResult = await fetchData();
-      if (typeof cacheResult !== 'undefined') {
-        if (this.isValueWithOptions(cacheResult)) {
-          ({ value } = cacheResult);
-          const { options: cacheResultOptions = {} } = cacheResult;
-          // Merging cache options from the "fetchData" result and passed method argument
-          options = Object.assign({}, options, cacheResultOptions);
-        } else {
-          value = cacheResult;
-        }
-
-        if (typeof value !== 'undefined') {
-          await this.set(key, value, options as SetCacheOptions);
-        }
-      }
-    }
+    this.getMap.set(key, this.rawGet(key, setOptions));
```

Comment:
I think there is too much `get` 🤔 
```suggestion
    this.pendingRequests.set(key, this.createRequest(key, setOptions));
```

Reply:
updated